js/app.jsx

```diff
@@ -1366,14 +1366,18 @@ SearchableCaseSelectionList = React.createClass({
     }
   },
 
- getInitialState: function() {
+  componentDidMount: function() {
+    this.loadOnePage(this.buildURL(this.props.query));
+  },
+
+  getInitialState: function() {
     return ({disableQueryURL: true, loaded: false});
   },
 
   render: function() {
     return (
       <div id={this.props.id}>
-        <SearchForm ref="searchform" parentId={this.props.id} disableQueryURL={this.state.disableQueryURL} query={this.state.query} onSubmit={this.handleSearch} syntaxlink={"help/syntax_caseselection.html"}/>
+        <SearchForm ref="searchform" parentId={this.props.id} disableQueryURL={this.state.disableQueryURL} query={this.props.query} onSubmit={this.handleSearch} syntaxlink={"help/syntax_caseselection.html"}/>
```

Comment:
If you use `this.props.query` here, will the `this.setState()` in line 1420 work?